src/main/java/com/ticketbooking/service/ReservationService.java

```diff
@@ -0,0 +1,323 @@
+package com.ticketbooking.service;
+
+import com.ticketbooking.dto.*;
+import com.ticketbooking.exception.TicketBookingNotFoundException;
+import com.ticketbooking.model.*;
+import com.ticketbooking.repo.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.MessageSource;
+import org.springframework.context.i18n.LocaleContextHolder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.mail.MessagingException;
+import javax.servlet.http.HttpServletRequest;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.*;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+@Service
+public class ReservationService {
+
+    private static final String DATE_FORMATTER= "yyyy-MM-dd HH:mm";
+
+    private MessageSource messageSource;
+    private MailService mailService;
+
+    private MovieRepo movieRepo;
+    private ScreeningRepo screeningRepo;
+    private RoomRepo roomRepo;
+    private SeatRepo seatRepo;
+    private TicketRepo ticketRepo;
+    private ReservationRepo reservationRepo;
+
+    @Autowired
+    public ReservationService(MessageSource messageSource, MovieRepo movieRepo, ScreeningRepo screeningRepo,
+                              ReservationRepo reservationRepo, RoomRepo roomRepo,
+                              SeatRepo seatRepo, TicketRepo ticketRepo, MailService mailService) {
+        this.movieRepo = movieRepo;
+        this.screeningRepo = screeningRepo;
+        this.roomRepo = roomRepo;
+        this.seatRepo = seatRepo;
+        this.ticketRepo = ticketRepo;
+        this.reservationRepo = reservationRepo;
+        this.mailService = mailService;
+        this.messageSource = messageSource;
+    }
+
+    @Transactional(readOnly = true)
+    public List<MovieDto> getMovies(String dateFrom, String dateTo) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(DATE_FORMATTER);
+        LocalDateTime formatDateFrom = LocalDateTime.parse(dateFrom, formatter);
+        LocalDateTime formatDateTo = LocalDateTime.parse(dateTo, formatter);
+
+        List<Screening> screenings = screeningRepo.findByScreeningTimeBetween(formatDateFrom, formatDateTo);
+
+        List<MovieDto> movies = screenings.stream()
+                .map(s -> {
+                    Optional<Movie> movie = movieRepo.findById(s.getMovie().getId());
+                    if (movie.isPresent()) {
+                        MovieDto movieDto = new MovieDto(movie.get().getId(), s.getId(), movie.get().getTitle(), s.getScreeningTime());
+                        return movieDto;
+                    } else {
+                        return null;
+                    }
+                })
+                .filter(m -> m != null)
```

Comment:
Szczerze to nie rozumiem. Może być taka sytuacja?

Reply:
Tak ta linijka jest tutaj nadmiarowa, taka sytuacja nie powinna się wydarzyć.